examples/ieciidc-listener.c

```diff
@@ -0,0 +1,440 @@
+	memcpy(entry->mpegts_packet, mpeg_tsp, MPEG_TS_PACKET_LEN);
+
+	STAILQ_INSERT_TAIL(&packets, entry, entries);
+
+	/* If this was the first entry inserted onto the queue, we need to arm
+	 * the timer.
+	 */
+	if (STAILQ_FIRST(&packets) == entry) {
+		int res;
+
+		res = arm_timer(fd, tspec);
+		if (res < 0) {
+			STAILQ_REMOVE(&packets, entry, packet_entry, entries);
+			free(entry);
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+static bool is_valid_packet(struct avtp_stream_pdu *pdu)
+{
+	struct avtp_common_pdu *common = (struct avtp_common_pdu *) pdu;
+	uint64_t val64;
+	uint32_t val32;
+	int res;
+
+	res = avtp_pdu_get(common, AVTP_FIELD_SUBTYPE, &val32);
+	assert(res == 0);
+	if (val32 != AVTP_SUBTYPE_61883_IIDC) {
+		fprintf(stderr, "Subtype mismatch: expected %u, got %u\n",
+						AVTP_SUBTYPE_61883_IIDC, val32);
+		return false;
+	}
+
+	res = avtp_pdu_get(common, AVTP_FIELD_VERSION, &val32);
+	assert(res == 0);
+	if (val32 != 0) {
+		fprintf(stderr, "Version mismatch: expected %u, got %u\n",
+								0, val32);
+		return false;
+	}
+
+	res = avtp_ieciidc_pdu_get(pdu, AVTP_IECIIDC_FIELD_TV, &val64);
+	assert(res == 0);
+	if (val64 != 0) {
+		fprintf(stderr, "tv mismatch: expected %u, got %" PRIu64 "\n",
+								0, val64);
+		return false;
+	}
+
+	res = avtp_ieciidc_pdu_get(pdu, AVTP_IECIIDC_FIELD_STREAM_ID,
+									&val64);
+	assert(res == 0);
+	if (val64 != STREAM_ID) {
+		fprintf(stderr, "Stream ID mismatch: expected %" PRIu64
+				", got %" PRIu64 "\n", STREAM_ID, val64);
+		return false;
+	}
+
+	res = avtp_ieciidc_pdu_get(pdu, AVTP_IECIIDC_FIELD_SEQ_NUM,
+									&val64);
+	assert(res == 0);
+
+	if (val64 != expected_seq) {
+		/* If we have a sequence number mismatch, we simply log the
+		 * issue and continue to process the packet. We don't want to
+		 * invalidate it since it is a valid packet after all.
+		 */
+		fprintf(stderr, "Sequence number mismatch: expected %u, got %"
+				PRIu64 "\n", expected_seq, val64);
+		expected_seq = val64;
+	}
+
+	expected_seq++;
+
+	res = avtp_ieciidc_pdu_get(pdu,
+				AVTP_IECIIDC_FIELD_STREAM_DATA_LEN, &val64);
+	assert(res == 0);
+	if (val64 != STREAM_DATA_LEN) {
+		fprintf(stderr, "Data len mismatch: expected %lu, got %"
+					PRIu64 "\n", STREAM_DATA_LEN, val64);
+		return false;
+	}
+
+	res = avtp_ieciidc_pdu_get(pdu, AVTP_IECIIDC_FIELD_TAG, &val64);
+	assert(res == 0);
+	if (val64 != AVTP_IECIIDC_TAG_CIP) {
+		fprintf(stderr, "tag mismatch: expected %u, got %" PRIu64 "\n",
+						AVTP_IECIIDC_TAG_CIP, val64);
+		return false;
+	}
+
+	res = avtp_ieciidc_pdu_get(pdu, AVTP_IECIIDC_FIELD_CHANNEL,
+									&val64);
+	assert(res == 0);
+	if (val64 != 31) {
+		fprintf(stderr, "channel mismatch: expected %u, got %" PRIu64
+							"\n", 31, val64);
+		return false;
+	}
+
+	res = avtp_ieciidc_pdu_get(pdu, AVTP_IECIIDC_FIELD_CIP_SID, &val64);
+	assert(res == 0);
+	if (val64 != 63) {
+		fprintf(stderr, "sid mismatch: expected %u, got %" PRIu64 "\n",
+							63, val64);
```

Comment:
Wrong code alignment. There are several other coding style issues like this in this file. Please review all the code.